freud/diffraction.pyx

```diff
@@ -0,0 +1,348 @@
+        inv_shear = np.linalg.inv(shear)
+        return inv_shear
+
+    def circle_cutout(self, p):
+        """
+        Find pixel indices in diffraction intensity array outside of the circle
+        Note: taken from Diffractometer.prep_sq()
+        Parameters
+        -------
+        p : numpy.ndarray (N,N), diffraction intensity array
+        Returns
+        -------
+        numpy.ndarray (N,), indices of particles outside the circle
+        note: N != to N in p.shape
+        """
+        y, x = np.indices(p.shape)
+        rmax = len(x) / 2 - 1
+        center = np.array([rmax, rmax])
+        # radii, constant for a single zoom
+        r = np.hypot(x - center[1], y - center[0]).flatten()
+        # array index into p corresponding to r
+        i = np.argsort(r.flat)
+        # sorted radius indices
+        r_sort = r.flat[i]
+        return i[r_sort > rmax]
+
+    def scale(self, a):
+        """
+        Scales up a matrix around middle particle
+        Note: Doesn't handle atoms on periodic boundaries perfectly --
+        intensity only on one half of boundary.
+        Parameters
+        -------
+        a : numpy.ndarray (N,N), input array
+        Returns
+        -------
+        numpy.ndarray (N,N), scaled array
+        """
+        ny, nx = np.shape(a)
+        y = np.array([list(range(ny))])
+        x = np.array([list(range(nx))])
+        d = interpolate.RectBivariateSpline(x, y, a, kx=1, ky=1)
+        x = np.linspace(0, nx, self.N)
+        y = np.linspace(0, ny, self.N)
+        d = d(x, y)
+        return d
+
+    def shear_back(self, img, inv_shear):
+        """
+        TODO
+        Parameters
+        ----------
+        img : numpy.ndarray (N,N), array of diffraction intensities
+        inv_shear : numpy.ndarray (2,2), inverse shear matrix
+        Returns
+        -------
+        numpy.ndarray (N,N), sheared array of diffraction intensities
+        """
+        roll = img.shape[0] / 2 - 1
+        ss = np.max(self.box) * inv_shear
+        A1 = np.array([[1, 0, -roll],
+                      [0, 1, -roll],
+                      [0, 0, 1]])
+
+        A2 = np.array([[ss[1, 0], ss[0, 0], roll],
+                      [ss[1, 1], ss[0, 1], roll],
+                      [0, 0, 1]])
+
+        A3 = np.linalg.inv(np.dot(A2, A1))
+        A4 = A3[0:2, 0:2]
+        A5 = A3[0:2, 2]
+        img = ndimage.interpolation.affine_transform(img, A4, A5,
+                                                     mode="constant")
+        return img
+
+    def compute(self, system, rot, cutout=True):
+        """
+        2D FFT to get diffraction pattern from intensity matrix.
+        Parameters
+        ----------
+        rot: numpy.ndarray (3, 3), rotation matrix
+        cutout: bool, return diffraction pattern with circle cutout
+        (default True)
+        Returns
+        -------
+        numpy.ndarray (N,N), diffraction pattern
+        """
+        box, positions = system
```

Comment:
This will only work if the system is provided as a tuple. You should pre-process the system with the `_make_default_nq` function and then use the NeighborQuery properties for box and points.

https://github.com/glotzerlab/freud/blob/e4969d50e979ec1ec8d29af0fb62c2a48d93e83f/freud/locality.pyx#L707